### PR TITLE
Add general polder deployment

### LINKED ISSUE #13: Polder Webapp helm chart
Have a dedicated helm chart to deploy the polder webapp. It will be incorporated later in a larget polder chart for deploying all the components.

### PR DESCRIPTION
**Pull request by pantierra**
This PR introduces a general `polder` chart that has as dependencies the `webapp-polder` and `eoapi`. It also fixes the failing tests of the CI pipeline.

Follow-up to #15 

Closes #13

### CODE REVIEW

File: deployment/helm/.gitignore [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# General files for the project
```
Comment (ciaransweet) [inline deployment/helm/.gitignore]: Do we need another `.gitignore` if we already have a project level one? Can this be merged to the top level one?

*edit* I'm aware this is autogen'd by helm

File: deployment/helm/polder/values.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+# This will set the replicaset count more information can be found here: https://kubernetes.io/docs/concepts/workloads/controllers/replicaset/
+replicaCount: 1
+
+# This is to override the chart name.
+nameOverride: "polder"
+fullnameOverride: "test-polder"
+
+# polder custom configuration
+mapboxToken: "pk.eyJ0IjoiZGV2c2VlZCIsImEiOiJnUi1mbkVvIn0.018aLhX0Mb0tdtaT2QNe2Q"
```
Comment (ciaransweet) [inline deployment/helm/polder/values.yaml]: Whatever this token is. We should definitely remove it *and* regenerate it on Mapbox... Given this repo is public, I assume this is now exposed and vulnerable

Comment (pantierra) [reply deployment/helm/polder/values.yaml]: Oh, I am sorry. I was assuming this was just to be some dummy token.

Shall we handle normal k8s secrets instead? And share the file in some private repo.

File: deployment/helm/.gitignore (hunk 1 above)
Comment (pantierra) [reply deployment/helm/.gitignore]: You are right. I also dislike having several `.gitignore` files. I just merged it with the root level `.gitignore`

**Review by ciaransweet — APPROVED** [review]

File: deployment/helm/polder/values.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+# This will set the replicaset count more information can be found here: https://kubernetes.io/docs/concepts/workloads/controllers/replicaset/
+replicaCount: 1
+
+# This is to override the chart name.
+nameOverride: "polder"
+fullnameOverride: "test-polder"
+
+# polder custom configuration
+mapboxToken: ""
+stacApi: "https://rx2hna9pbg.execute-api.eu-central-1.amazonaws.com"
```
Comment (emmanuelmathot) [inline deployment/helm/polder/values.yaml]: Those are links for WHIS, I would replace them with the default URL generated by the default helm deploy on kubernetes. (e.g. polder-eopi-stac)

**Review by emmanuelmathot — CHANGES_REQUESTED** [review]

**Comment by emmanuelmathot** [comment]
I updated ther workflow to make the k8s integration test working but there are still issue when deploying eoapi-k8s

**Comment by emmanuelmathot** [comment]
There is [something wrong with the vector API in eoAPI](https://github.com/developmentseed/polder/actions/runs/11701112077/job/32586512382?pr=16#step:14:4451) that is not captured by the eoAPI helm chart and thus gives a false positive. A readiness probe probably needs to be set up correctly.

File: deployment/helm/polder/values.yaml (hunk 3 above)
Comment (pantierra) [reply deployment/helm/polder/values.yaml]: Done.

**Comment by pantierra** [comment]
Many thanks for the review! I adjusted the urls, and created a follow up ticket for the additional tests #17.

**Review by emmanuelmathot — APPROVED** [review]

**Pull request merged**